src/sentry/tasks/merge.py

```diff
@@ -228,6 +229,50 @@ def rehash_group_events(group_id, transaction_id=None, **kwargs):
     delete_group.delay(group.id)
 
 
+def decrement_new_issue_count(group):
+    """
+    Updates the new issue/groups count
+
+    Should be called with the old group from merge_group.
+    That will be deleted, and so the count is decremented by 1.
+    """
+    from sentry.models import GroupRelease, Release, ReleaseProject, ReleaseProjectEnvironment
+    release_version = group.get_first_release()
+    if release_version is None:
+        return
+    try:
+        release = Release.objects.get(version=release_version)
+    except Release.DoesNotExist:
+        return
+    release.update(new_groups=release.new_groups - 1)
+
+    try:
+        release_project = ReleaseProject.objects.get(
+            release_id=release.id,
+            project_id=release.project_id,
+        )
+    except ReleaseProject.DoesNotExist:
+        # Done updating can't have a ReleaseProjectEnvironment without a ReleaseProject
```

Comment:
Was this `try`/ `except` based on a specific scenario that you expect to occur? Or is this just being careful?

Reply:
Great question. I believe it's me being careful. I don't tend to trust that things will be there unless there's a specific database constraint.